services/harmony/app/frontends/service-image-tags.ts

```diff
@@ -0,0 +1,156 @@
+
+import { Response, NextFunction } from 'express';
+import HarmonyRequest from '../models/harmony-request';
+import { getEdlGroupInformation } from '../util/edl-api';
+
+const harmonyTaskServices = [
+  'work-item-scheduler',
+  'work-item-updater',
+  'work-reaper',
+  'work-failer',
+];
+
+/**
+ * Compute the map of services to tags. Harmony core services are excluded.
+ * @returns The map of canonical service names to image tags.
+ */
+function getImageTagMap(): {} {
+  const imageMap = {};
+  for (const v of Object.keys(process.env)) {
+    if (v.endsWith('_IMAGE')) {
+      const serviceName = v.slice(0, -6).toLowerCase().replaceAll('_', '-');
+      // add in any services that are not Harmony core task services
+      if (!harmonyTaskServices.includes(serviceName)) {
+        const image = process.env[v];
+        const tag = image.match(/.*:(.*)/)[1] || '';
+        imageMap[serviceName] = tag;
+      }
+    }
+  }
+
+  return imageMap;
+}
+
+/**
+ * Validate that the service exists
+ * @param req - The request object
+ * @param res  - The response object - will be used to send an error if the validation fails
+ * @returns A Promise containing `true` if the service exists, `false` otherwise
+ */
+async function validateServiceExists(
+  res: Response, service: string,
+): Promise<boolean> {
+  const imageMap = getImageTagMap();
+  if (!imageMap[service]) {
+    res.statusCode = 404;
+    const message = `Service ${service} does not exist.\nThe existing services and their images are\n${JSON.stringify(imageMap, null, 2)}`;
+    res.send(message);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Validate that the user is in the deployers or the admin group
+ * @param req - The request object
+ * @param res  - The response object - will be used to send an error if the validation fails
+ * @returns A Promise containing `true` if the user in in either group, `false` otherwise
+ */
+async function validateUserIsInDeployerOrAdminGroup(
+  req: HarmonyRequest, res: Response,
+): Promise<boolean> {
+  const { isAdmin, isServiceDeployer } = await getEdlGroupInformation(
+    req.user, req.context.logger,
+  );
+
+  if (!isServiceDeployer && !isAdmin) {
+    res.statusCode = 403;
+    res.send(`User ${req.user} is not in the service deployers or admin EDL groups`);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Verify that the given tag is valid. Send an error if it is not.
+ * @param req - The request object
+ * @param res  - The response object - will be used to send an error if the validation fails
+ * @returns a Promise containing `true` if the tag is valid, false if not
+ */
+async function validateTag(
+  req: HarmonyRequest, res: Response,
+): Promise<boolean> {
+  const { tag } = req.body;
+  const tagRegex = /^[a-zA-Z\d_][a-zA-Z\d\-_.]{0,127}$/;
```

Comment:
I think you mentioned finding the tag requirements documented somewhere - maybe add a link to the documentation.

Reply:
done